sentry/src/main/java/io/sentry/SentryClient.java

```diff
@@ -146,6 +149,13 @@ private boolean shouldApplyScopeData(
       }
     }
 
+    final boolean shouldSendSessionUpdate =
+        shouldSendSessionUpdateForDroppedEvent(sessionBeforeUpdate, session);
+
+    if (event == null && !shouldSendSessionUpdate) {
+      return SentryId.EMPTY_ID;
```

Comment:
maybe add a debug log here as of why it's been dropped?

Reply:
Debug log for the event itself is already in place further up. This early return would mean we don't want to send the session update alone as it wasn't a session health change.

Do you mean like this?
```suggestion
    options
                .getLogger()
                .log(
                    SentryLevel.DEBUG,
                    "Not sending session update for dropped event %s as it did not cause the session health to change.",
                    event.getEventId());
      return SentryId.EMPTY_ID;
```

Reply:
Yeah, exactly  👍

Reply:
Though the `event` is null here, so `event.getEventId()` wouldn't work

Reply:
Will make this compile, just wanted to use the suggestion as basis for discussion.